var/spack/repos/builtin/packages/rivet/package.py

```diff
@@ -106,16 +106,16 @@ class Rivet(AutotoolsPackage):
     depends_on('yoda@1.7.0:1.7', when='@2.6.0,2.7.0,2.7.1,3.0.0,3.0.2')
     depends_on('yoda@1.5.0:1.5', when='@2.4.1')
 
-    depends_on('hepmc',  type=('build', 'link', 'run'), when='hepmc=2')
-    depends_on('hepmc3', type=('build', 'link', 'run'), when='hepmc=3')
-    depends_on('boost', when='@:2.5.0', type=('build', 'run'))
-    depends_on('fastjet', type=('build', 'run'))
-    depends_on('fjcontrib', type=('build', 'run'), when='@3.0.0:')
-    depends_on('gsl', type=('build', 'run'), when='@:2.6.0,2.6.2:2')
+    depends_on('hepmc', when='hepmc=2')
+    depends_on('hepmc3', when='hepmc=3')
+    depends_on('boost', when='@:2.5.0')
+    depends_on('fastjet')
+    depends_on('fjcontrib', when='@3.0.0:')
+    depends_on('gsl', when='@:2.6.0,2.6.2:2')
     depends_on('python', type=('build', 'run'))
     depends_on('py-cython@0.24.0:', type='build')
-    depends_on('swig', type=('build', 'run'))
-    depends_on('yaml-cpp', when='@2.0.0:2.1.2', type=('build', 'run'))
+    depends_on('swig')
```

Comment:
SWIG shouldn't ever  be a runtime dependency.

Reply:
done, thanks.